plugins/security/nexus-ldap-plugin/src/main/java/org/sonatype/security/realms/ldap/internal/persist/DefaultLdapConfigurationManager.java

```diff
@@ -201,43 +191,47 @@ private void mayActivateLdapRealm() throws InvalidConfigurationException {
    *
    * @since 2.7.0
    */
-  private void mayDeactivateLdapRealm() throws InvalidConfigurationException {
-    final List<String> activeRealms = securitySystem.getRealms();
-    if (activeRealms.contains(LdapPlugin.REALM_NAME)) {
-      activeRealms.remove(LdapPlugin.REALM_NAME);
-      securitySystem.setRealms(activeRealms);
+  private void mayDeactivateLdapRealm() {
+    try {
+      final List<String> activeRealms = securitySystem.getRealms();
+      if (activeRealms.contains(LdapPlugin.REALM_NAME)) {
+        activeRealms.remove(LdapPlugin.REALM_NAME);
+        securitySystem.setRealms(activeRealms);
+      }
+    }
+    catch (Exception e) {
+      throw Throwables.propagate(e);
     }
   }
 
-  private CLdapConfiguration getConfiguration() {
-    if (ldapConfiguration == null) {
+  /**
+   * Primes the cache if not primed and returns it, so to say "lazily" loads. Should be called only from synchronized
+   * method.
+   */
+  private LinkedHashMap<String, LdapConfiguration> getConfiguration() {
+    if (!cachePrimed) {
       try {
-        final CLdapConfiguration config = configurationSource.load();
-        final ValidationResponse vr = configurationValidator
-            .validateModel(new ValidationRequest<CLdapConfiguration>(config));
-        if (vr.getValidationErrors().size() > 0) {
-          throw new InvalidConfigurationException(vr);
+        cache.clear();
+        final List<LdapConfiguration> ldapConfigurations = configurationSource.loadAll();
+        Collections.sort(ldapConfigurations, new Comparator<LdapConfiguration>()
+        {
+          @Override
+          public int compare(final LdapConfiguration o1, final LdapConfiguration o2) {
+            return o1.getOrder() - o2.getOrder();
+          }
+        });
+        for (LdapConfiguration ldapConfiguration : ldapConfigurations) {
+          cache.put(ldapConfiguration.getId(), ldapConfiguration);
         }
-        ldapConfiguration = config;
       }
-      catch (ConfigurationException e) {
-        log.error("Invalid LDAP Configuration", e);
+      catch (Exception e) {
+        log.warn("Cannot retrieve LDAP configuration", e);
```

Comment:
Makes sense, and reasonable considering the level of 'log spam' we're talking about in NX.